boot.js

```diff
@@ -114,34 +119,34 @@ function Boot (server, opts, done) {
   }
   this._thereIsCloseCb = false
 
-  // we init, because we need to emit "start" if no use is called
-  this._init()
+  this._doStart = null
+  const main = new Plugin(this, (s, opts, done) => {
+    this._doStart = done
+    if (opts.autostart) {
+      this.start()
+    }
+  }, opts, noop)
+
+  Plugin.loadPlugin.call(this, main, (err) => {
+    debug('root plugin ready')
+    if (err) {
+      this._error = err
+      if (this._readyQ.length() === 0) {
+        throw err
+      }
+    } else {
+      this._readyQ.resume()
+    }
+  })
 }
 
 inherits(Boot, EE)
 
-Boot.prototype._init = function () {
-  if (this.booted) {
-    throw new Error('root plugin has already booted')
-  }
+Boot.prototype.start = function () {
```

Comment:
I don’t think we should add it to wrap? Why would you want to do it?

An avvio instance can be booted only once. Otherwise too many edge cases happens (this is discussed elsewhere).

Reply:
Nevermind I looked at the docs again it explicitly states which functions are exposed.  It needs to be updated to list `close` and `onClose` but that's not related to this review.  I'll submit a doc update for that later.